drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/InvocationResponseFuture.java

```diff
@@ -73,6 +79,25 @@ private InvocationResponseFuture(InvokeRequest request, ConnectionParameters con
     private synchronized void tryConnect()
     {
         try {
+            Optional<Boolean> encryptionRequired = request.getAddress().isEncryptionRequired();
+            if (encryptionRequired.isPresent() && encryptionRequired.get() && !connectionParameters.getSslContextParameters().isPresent()) {
+                fatalError(new TException(format(
```

Comment:
Once you move this algorithm to the `createInvocationResponseFuture` you can simply throw this exception.